static/app/components/replays/highlightReplayPlugin.tsx

```diff
@@ -9,77 +9,151 @@ interface HighlightReplayPluginOptions {
+    if (
+      this.highlightsByNodeId.size &&
+      'removes' in event.data &&
+      event.data.removes?.length
+    ) {
+      // If a highlighted node was removed from DOM we need to remove the highlight canvas
+      event.data.removes.forEach(removedNode => {
+        if ('id' in removedNode && this.highlightsByNodeId.has(removedNode.id)) {
+          this.removeHighlight(removedNode.id, replayer);
+        }
+      });
+    }
 
-      const ctx = canvas.getContext('2d');
+    // Check for expired highlights
+    if (this.highlightsByNodeId.size) {
+      Array.from(this.highlightsByNodeId.entries())
+        .filter(
+          ([, {added, expires}]) => event.timestamp < added || event.timestamp >= expires
+        )
+        .forEach(([nodeId]) => {
+          this.removeHighlight(nodeId, replayer);
+        });
+    }
 
-      if (!ctx) {
-        return;
-      }
+    // If we have highlights and the replay event is a scroll event, update the highlight
+    // `event.type` == EventType.IncrementalSnapshot (https://github.com/rrweb-io/rrweb/blob/master/packages/rrweb/src/types.ts#L20)
+    // `event.data.source` == IncrementalSource.Scroll (https://github.com/rrweb-io/rrweb/blob/master/packages/rrweb/src/types.ts#L83)
+    if (this.highlightsByNodeId.size && event.type === 3 && event.data.source === 3) {
+      // TODO(replays): Re-draw rect instead of removing it (:
+      Array.from(this.highlightsByNodeId.entries()).forEach(([nodeId]) => {
+        this.removeHighlight(nodeId, replayer);
+      });
+    }
 
-      // TODO(replays): Does not account for scrolling (should we attempt to keep highlight visible, or does it disappear)
+    if (!isHighlightEvent(event) || event.data.nodeId < 0) {
+      return;
+    }
 
-      // Draw a rectangle to highlight element
-      ctx.fillStyle = highlightColor;
-      ctx.fillRect(left, top, width, height);
+    const highlightObj = event.data;
 
-      // Draw a dashed border around highlight
-      ctx.beginPath();
-      ctx.setLineDash([5, 5]);
-      ctx.moveTo(left, top);
-      ctx.lineTo(left + width, top);
-      ctx.lineTo(left + width, top + height);
-      ctx.lineTo(left, top + height);
-      ctx.closePath();
-      ctx.stroke();
+    // @ts-expect-error mouseTail is private
+    const {mouseTail, wrapper} = replayer;
+    const mirror = replayer.getMirror();
+    const node = mirror.getNode(highlightObj.nodeId);
 
-      ctx.font = '24px Rubik';
-      ctx.textAlign = 'right';
-      ctx.textBaseline = 'bottom';
+    // TODO(replays): There is some sort of race condition here when you "rewind" a replay,
+    // mirror will be empty and highlight does not get added because node is null
+    if (!node || !replayer.iframe.contentDocument?.body?.contains(node)) {
+      return;
+    }
 
-      const textWidth = ctx.measureText(highlightObj.text).width;
+    // @ts-ignore This builds locally, but fails in CI -- ignoring for now
```

Comment:
@ryan953 stuck this in here... not sure why it builds fine locally. Even wiped node_modules.